backend/httpd/main.go

```diff
@@ -1,14 +1,33 @@
 package main
 
-import "github.com/gin-gonic/gin"
+import (
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+)
 
 func main() {
 	r := gin.Default()
+
+	// This allows for front end calls made from the origin source
+	// http://localhost:3000
+	// This will need to be changed/updated to either be "*", ie
+	// allow all origins (which could be bad for security), or
+	// the url for where ever we host our front end.
+	// All RESTful api methods are allowed
+	r.Use(cors.New(cors.Config{
+		AllowOrigins:     []string{"http://localhost:3000", "https://dialog-gamma.vercel.app/"},
```

Comment:
Great idea, do you think we should add the "preview" deployments as well?
Taking a look at the other PRs, their urls (which are generated by vercel) are:
https://dialog-git-pgcompose.white-van.vercel.app/

https://dialog-git-dev-036.white-van.vercel.app/

https://dialog-git-dev-031-b.white-van.vercel.app/

Reply:
hm, adding all the domains manually seems like a bad idea. I think CORS supports regex/pattern matching, so we could match all sites under the white-van.vercel.app subdomain?

https://www.moesif.com/blog/technical/cors/Authoritative-Guide-to-CORS-Cross-Origin-Resource-Sharing-for-REST-APIs/
https://stackoverflow.com/questions/14003332/access-control-allow-origin-wildcard-subdomains-ports-and-protocols

Reply:
I have added regex to the AllowOriginFunction which checks to see if the incoming origin matches the allowed URLs. It should allow for calls from http://localhost:3000, and any of the white-van.vercel.app/ subdomains.